dotnet/src/Extensions/PromptTemplate.Handlebars/Helpers/KernelFunctionHelpers/KernelFunctionHelpers.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using HandlebarsDotNet;
+using HandlebarsDotNet.Compiler;
+
+#pragma warning disable IDE0130 // Namespace does not match folder structure
+namespace Microsoft.SemanticKernel.PromptTemplate.Handlebars.Helpers;
+#pragma warning restore IDE0130
+
+/// <summary>
+/// Extension class to register Kernel functions as helpers.
+/// </summary>
+public static class KernelFunctionHelpers
+{
+    /// <summary>
+    /// Register all (default) or specific categories.
+    /// </summary>
+    /// <param name="handlebarsInstance">The <see cref="IHandlebars"/>-context.</param>
+    /// <param name="kernel">Kernel instance.</param>
+    /// <param name="executionContext">Kernel arguments maintained as the executing context.</param>
+    /// <param name="nameDelimiter">The character used to delimit the plugin name and function name in a Handlebars template.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    public static void Register(
+        IHandlebars handlebarsInstance,
+        Kernel kernel,
+        KernelArguments executionContext,
+        string nameDelimiter,
+        CancellationToken cancellationToken = default)
+    {
+        kernel.Plugins.GetFunctionsMetadata().ToList()
+            .ForEach(function =>
+                RegisterFunctionAsHelper(kernel, executionContext, handlebarsInstance, function, nameDelimiter, cancellationToken)
+            );
```

Comment:
This should just be a foreach loop, rather than allocating a List and ForEach'ing that, e.g.
```C#
foreach (var function in kernel.Plugins.GetFunctionsMetadata())
{
    RegisterFunctionAsHelper(kernel, executionContext, handlebarsInstance, function, nameDelimiter, cancellationToken);
}
```

Also, this method is taking a CancellationToken but it's not async... that's an unusual combination outside of certain domains.  What's the purpose of the CancellationToken here?

Reply:
`RegisterFunctionAsHelper` calls `InvokeKernelFunctionAsync`, which is where the cancellation token is actually used. 

We use a GetAwaiter there since Handlebars doesn't support async ops yet